Limit Max Node Graph Zoom

danieldresser-ie opened the pull request:
This ended up being a bit more philosophically complex than I expected, because of the existing tension in ViewportGadget between being used as a 3D viewport and being used for the node graph.  Basically all of the existing public interface applies to either the planar or non-planar mode.  I started out trying to match this, with a V3f minimumViewSize that could act to limit either how close a perspective camera can get to it's centerOfInterest, or how much a planar camera can zoom in.  But trying to work out the details, in ended up making everything a lot more complicated - in particular, we wanted to match the way that the `focus` function sizes things based on a pixel size, but to do this requires a dynamic calculation where GraphEditor compares it's current pixel size to the size of the bound, creates a bound in graph units, and then ViewportGadget converts back to a pixel scaling.  If we took the same approach to the max zoom level, this would mean that max zoom level would need to be recomputed if the size of the GraphEditor changes.  I got as far as working out the ViewportGadget side here: https://github.com/danieldresser-ie/gaffer/tree/zoomLimitBoundsBased

... but then gave up and did a much simpler version here.  Still not sure about the right approach ... see what you think of this.

johnhaddon commented:
This works great for me, and I think limiting it to planar mode is a good call. I'm a little fuzzy on the definition of maxPlanarZoom, so maybe you educate me in our call today...

danieldresser-ie commented:
Fixed completely outdated comment describing setMaxPlanarZoom, added Changes entries, and changed target to 0.59_maintenance, should be good to merge now.
